origins_common/inc/layout_builder_search.inc

```diff
@@ -0,0 +1,235 @@
+      $field = FieldConfig::create([
+        'field_storage' => $field_storage,
+        'bundle' => $entity->id(),
+        'label' => t('LB Search enabled'),
+      ]);
+      $field->setTranslatable(FALSE);
+      $field->save();
+    }
+
+    $field = FieldConfig::loadByName('node', $entity->id(), 'field_lb_search_content');
+    if (!$field) {
+      // Create the content field for this entity.
+      $field_storage = FieldStorageConfig::loadByName('node', 'field_lb_search_content');
+      $field = FieldConfig::create([
+        'field_storage' => $field_storage,
+        'bundle' => $entity->id(),
+        'label' => t('LB Search content'),
+        'description' => t('Populated on node save.'),
+      ]);
+      $field->setTranslatable(FALSE);
+      $field->save();
+    }
+
+    // Enable the fields in the form display.
+    $form_display = \Drupal::entityTypeManager()->getStorage('entity_form_display')->load('node.' . $entity->id() . '.default');
+    $form_display->setComponent('field_lb_search_enable', [
+      'type' => 'boolean_checkbox',
+      'label' => 'above',
+      'settings' => ['link_to_entity' => 'false'],
+    ]);
+
+    $form_display->setComponent('field_lb_search_content', [
+      'type' => 'string_textarea',
+      'label' => 'above',
+      'settings' => ['link_to_entity' => 'false'],
+    ])->save();
+
+  }
+  else {
+    // Delete the 'enable search index' field from this entity if it exists.
+    $field = FieldConfig::loadByName('node', $entity->id(), 'field_lb_search_enable');
+    if ($field) {
+      $field->delete();
+    }
+
+    $field = FieldConfig::loadByName('node', $entity->id(), 'field_lb_search_content');
+    if ($field) {
+      $field->delete();
+    }
+  }
+}
+
+/**
+ * Renders and stores the layout builder display to the search index field.
+ *
+ * @param \Drupal\node\Entity\Node $entity
+ *   The entity to process.
+ */
+function layout_builder_search_node_presave(Node $entity) {
+  if ($entity->hasField('field_lb_search_enable')) {
+    $search_enable = (bool) $entity->get('field_lb_search_enable')->getValue();
+    if ($search_enable) {
+      $entity_storage = \Drupal::entityTypeManager()->getStorage('node_type');
+      $entity_type = $entity_storage->load($entity->getType());
+      $settings = $entity_type->getThirdPartySettings('origins_common');
+      $search_index_display = $settings['lb_search_index_display'] ?? NULL;
+
+      if ($search_index_display) {
+        // Check the requested search index storage field exists.
+        if ($entity->hasField('field_lb_search_content')) {
+          $view_builder = \Drupal::entityTypeManager()->getViewBuilder('node');
+          $build = $view_builder->view($entity, $search_index_display);
+          // Ideally we would use use a defined text format to filter
+          // the content, this will suffice for the time being.
+          $content = strip_tags(\Drupal::service('renderer')->render($build));
```

Comment:
I had some trouble in the past with strip_tags encoding things like ampersands, which might translate to things being stored in odd ways in Solr. Worth keeping this in for now and I guess it can be relaxed/replaced with something more suitable if/when any trouble arises.